components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManager.java

```diff
@@ -771,21 +771,30 @@ private void updatePermissions(String roleId, List<PatchOperation> permissionOpe
 
     private void prepareAddedRemovedGroupLists(Set<String> addedGroupsIds, Set<String> removedGroupsIds,
                                                Set<String> replacedGroupsIds, PatchOperation groupOperation,
-                                               Map<String, String> groupObject, List<GroupBasicInfo> groupListOfRole) {
+                                               Map<String, String> groupObject, List<GroupBasicInfo> groupListOfRole)
+            throws BadRequestException {
+
+        String value = groupObject.get(SCIMConstants.CommonSchemaConstants.VALUE);
+
+        if (StringUtils.isBlank(value)) {
+            throw new BadRequestException(
+                    "Updating groups of the role by display name is not supported. Update using group id instead.",
```

Comment:
```suggestion
                    "Group id is required to update group of the role.",
```

Let's make the error message generic to all cases come here.
Fix other place as well

Reply:
Sure. Resolved.